File: test_data/csharp/test_main/aas_core_meta.v3rc2/input/snippets/Verification/qualifier_types_are_unique.cs
/// <summary>
/// Check that there are no duplicate <see cref="Aas.Qualifier.Type" />'s
/// in the <paramref name="qualifiers" />.
/// </summary>
public static bool QualifierTypesAreUnique(
    IEnumerable<Aas.Qualifier> qualifiers
)
{
    var typeSet = new HashSet<string>();
    foreach (var qualifier in qualifiers)
    {
        if (typeSet.Contains(qualifier.Type))
        {
            return false;
        }
        typeSet.Add(qualifier.Type);
    }
    return true;
}

File: test_data/golang/test_main/aas_core_meta.v3/input/snippets/Verification/qualifier_types_are_unique.go
// Check that there are no duplicate [aastypes.IQualifier.Type]'s in
// the `qualifiers`.
func QualifierTypesAreUnique[Q aastypes.IQualifier](
	qualifiers []Q) bool {
	typeSet := make(map[string]struct{})
	for _, qualifier := range qualifiers {
		t := qualifier.Type()

		_, has := typeSet[t]
		if has {
			return false
		}
		typeSet[t] = struct{}{}
	}
	return true
}
